src/createMapper.ts

```diff
@@ -1,21 +1,43 @@
-import { Mapper, Profile } from '@/types'
+import { ProfileNotFoundError } from '@/profileNotFoundError'
+import { Mapper, Profile, ProfileKey } from '@/types'
 
 export function createMapper<T extends Profile>(profiles: T[]): Mapper<T> {
 
+  const profileMap = buildProfilesMap(profiles)
+
+  function findProfile(sourceKey: T['sourceKey'], destinationKey: T['destinationKey']): Profile {
```

Comment:
@stackoverfloweth nit but I wanna change this to `getProfile` because `Map.get` vs `array.find`. Find implies searching an array IMO.

Reply:
True. I chose find because we don't know for sure that it will exist. I'm good either way though 👍